staging/src/k8s.io/apimachinery/pkg/util/sort/sort.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sort
+
+import (
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// SortDiscoveryGroupsTopo performs a topological consensus sort of API discovery group names.
+//
+// This is needed in Kubernetes API discovery merging logic, where multiple peers (apiserver instances)
+// may report different groups. To present a consistent peer-aggregated discovery response,
+// we use this function to compute a consensus ordering that respects the relative orderings
+// reported by all peers. This is called from the peer-aggregated discovery handler when constructing
+// the final APIGroupDiscoveryList for clients.
+func SortDiscoveryGroupsTopo(peers [][]string) []string {
```

Comment:
I think the order of peers can affect the results, so should document that callers should pass in peers in a consistent order to get deterministic results, and when calling this, we should sort peers by something that all peers will have access to (identity name or something)